kani-compiler/src/kani_middle/transform/check_aliasing/instrumentation.rs

```diff
@@ -0,0 +1,446 @@
+        let created_ref = self.meta_stack.get(&created).unwrap();
+        let reference_ref = self.meta_stack.get(&reference).unwrap();
+        self.call(
+            Signature::new("KaniNewMutRawFromRef", &[GenericArgKind::Type(ty)]),
+            vec![*created_ref, *reference_ref],
+            self.unit,
+        )?;
+        Ok(())
+    }
+
+    /// Instrument each of the locals collected into values with
+    /// initialization data.
+    pub fn instrument_locals(&mut self) -> Result<()> {
+        for local in (self.body.arg_count() + 1)..self.local_count {
+            self.instrument_local(local)?
+        }
+        Ok(())
+    }
+
+    /// Fetch the actions to be instrumented at the current instruction.
+    pub fn instruction_actions(&self) -> Vec<Action> {
+        let mut visitor = CollectActions::new(self.body.locals());
+        match self.min_processed {
+            SourceInstruction::Terminator { .. } => { /* not yet handled */ }
+            SourceInstruction::Statement { idx, bb } => {
+                visitor.visit_statement(&self.body.blocks()[bb].statements[idx]);
+            }
+        }
+        visitor.finalize()
+    }
+
+    /// Instrument the action given in "action" with the appropriate
+    /// update to the stacked borrows state.
+    fn instrument_action(&mut self, action: Action) -> Result<()> {
+        match action {
+            Action::StackCheck => self.instrument_stack_check(),
+            Action::NewStackReference { lvalue, rvalue } => {
+                self.instrument_new_stack_reference(lvalue, rvalue)
+            }
+            Action::StackUpdateReference { place, ty } => {
+                self.instrument_stack_update_ref(place, ty)
+            }
+            Action::NewMutRefFromRaw { lvalue, rvalue, ty } => {
+                self.instrument_new_mut_ref_from_raw(lvalue, rvalue, ty)
+            }
+            Action::StackUpdatePointer { place, ty } => self.instrument_stack_update_ptr(place, ty),
+            Action::NewMutRawFromRef { lvalue, rvalue, ty } => {
+                self.instrument_new_mut_raw_from_ref(lvalue, rvalue, ty)
+            }
+        }
+    }
+
+    /// Instrument all of the instructions and terminators in the function body
+    /// with appropriate updates to the stacked borrows state
+    /// and with validity assertions on the stacked borrows state.
+    pub fn instrument_instructions(&mut self) -> Result<()> {
+        loop {
+            let actions = self.instruction_actions();
+            if actions.len() > 0 {
+                eprintln!("Instrumenting actions:");
+                self.process_instruction();
+            }
+            for action in actions {
+                eprintln!("Action is: {:?}", action);
+                self.instrument_action(action)?;
+            }
+            self.min_processed = match self.min_processed {
+                SourceInstruction::Statement { idx: 0, bb: 0 } => {
+                    break;
+                }
```

Comment:
Perhaps a better way to express this would be writing it out as something like `SourceInstruction::Statement { idx: 0, bb: 0 } | SourceInstruction::Terminator { bb: 0 } => { break; }`

Reply:
uh if you have bb0 you can still have instructions to process